File: ABC081B.c
/* ABC081B - Shift only */
#include <stdio.h>
#include <stdbool.h>
#include <stdint.h>
#include <string.h>

bool all(int64_t n, int64_t *a)
{
    for (int64_t i = 0; i < n; ++i)
    {
        if (a[i] & 1)
        {
            return false;
        }
    }
    return true;
}

void half(int64_t n, int64_t *a)
{
    for (int64_t i = 0; i < n; ++i)
    {
        a[i] = a[i] >> 1;
    }
}

int64_t solver(int64_t n, int64_t *a)
{
    int64_t counter = 0;
    while (all(n, a))
    {
        half(n, a);
        counter++;
    }
    return counter;
}

int main()
{
    static int64_t a[200];
    int64_t n;
    scanf("%ld", &n);
    for (int64_t i = 0; i < n; ++i)
    {
        scanf("%ld", &a[i]);
    }
    printf("%ld\n", solver(n, a));
    return 0;
}

File: ABC081B.ml
(* ABC081B - Shift only *)

let even x = x mod 2 = 0

let half x = x / 2

let rec solver = function
  | xs when List.for_all even xs -> 1 + solver (List.map half xs)
  | _ -> 0

let () =
  let _ = read_int () in
  read_line ()
  |> Str.split (Str.regexp " ")
  |> List.map int_of_string |> solver |> string_of_int |> print_endline
